vulnerabilities/scraper/archlinux.py

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright (c) 2017 nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/vulnerablecode/
+# The VulnerableCode software is licensed under the Apache License version 2.0.
+# Data generated with VulnerableCode require an acknowledgment.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with VulnerableCode or any VulnerableCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with VulnerableCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  VulnerableCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  VulnerableCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/vulnerablecode/ for support and download.
+
+import json
+from urllib.request import urlopen
+
+
+ARCHLINUX_TRACKER_URL = 'https://security.archlinux.org/json'
+
+
+def extract_vulnerabilities(arch_data):
+    """
+    Return a sequence of mappings for each existing combination of
+    package and vulnerability from a mapping of arch linux vulnerabilities
+    data.
+    """
+    for item in arch_data:
+        package_vulnerabilities = []
+        vulnerabilities = item['issues']
+        packages_name = item['packages']
+
+        if not vulnerabilities or not packages_name:
+            continue
+
+        for package in packages_name:
+            affected_version = item['affected']
+            fixed_version = item['fixed']
+            for vulnerability in vulnerabilities:
+                package_vulnerabilities.append({
+                    'package_name': package,
+                    'vulnerability_id': vulnerability,
+                    'description': item['type'],
+                    'status': item['status'],
+                    'severity': item['severity'],
+                    'version': affected_version
+                })
+
+            for vulnerability in vulnerabilities:
+                if not fixed_version:
+                    fixed_version = ''
```

Comment:
Why would you care about an empty string vs. a None?

Reply:
Actually `fixed_version` attribute contains `null` value which on dumping in database throws error so I thought to convert the `null` into an `empty string` before dumping into the database.